polkadot/node/core/pvf/common/src/executor_intf.rs

```diff
@@ -106,8 +109,9 @@ pub fn params_to_wasmtime_semantics(par: &ExecutorParams) -> Result<Semantics, S
 	for p in par.iter() {
 		match p {
 			ExecutorParam::MaxMemoryPages(max_pages) =>
-				sem.heap_alloc_strategy =
-					HeapAllocStrategy::Dynamic { maximum_pages: Some(*max_pages) },
+				sem.heap_alloc_strategy = HeapAllocStrategy::Dynamic {
+					maximum_pages: Some((*max_pages).saturating_add(DEFAULT_HEAP_PAGES_ESTIMATE)),
```

Comment:
Nice! But can we just use the value from `sem` instead?

Reply:
I think we discussed it [here](https://github.com/paritytech/polkadot-sdk/pull/1774#discussion_r1343770041)

Reply:
Oh I misread this code, nevermind. Should it be this instead though?

```suggestion
					maximum_pages: Some((*max_pages).saturating_add(EXTRA_HEAP_PAGES)),
```

And are these values correct? I would have thought the extra pages were 32. @s0me0ne-unkn0wn 

```rs
const DEFAULT_HEAP_PAGES_ESTIMATE: u32 = 32;
const EXTRA_HEAP_PAGES: u32 = 2048;
```

Reply:
No, I believe everything is okay with the current version.

`EXTRA_HEAP_PAGES` is the number of pages of linear memory the instance is allowed to allocate in addition to its initial memory size.

`DEFAULT_HEAP_PAGES_ESTIMATE` is the number of pages the instance will use for its own purposes, they are not part of linear memory.

So, the corresponding executor parameter defines the number of linear memory pages the instance is allowed to allocate, and here, we implicitly add those auxiliary pages to that limit (for example, we can set the limit to 0, and Wasm code not using any linear memory should still be functional, but it's not possible without having some space for shadow stack and other bits).

Reply:
Would it clarify stuff if we name it `SHADOW_STACK_HEAP_PAGES`? Then current name may be a bit vague.

Reply:
It's not just shadow stack, according to the comment :thinking: But I doubt if the reasoning behind the calculation provided is correct enough. I didn't dig deeper into this aspect of Wasmtime, but it seems obvious that data segment pages shouldn't be allocated implicitly. Wasm source must specify the initial memory size, and it should always be enough to accommodate the data segment at least. Considering that I'm not the author of the original constant naming, I'm not 100% sure what is better, but probably renaming `DEFAULT_HEAP_PAGES_ESTIMATE` to `INITIAL_HEAP_PAGES_ESTIMATE` would bring more clarity.